src/Security/GoogleAuthenticator.php

```diff
@@ -0,0 +1,167 @@
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+class GoogleAuthenticator extends SocialAuthenticator
+{
+    /**
+     * @var ClientRegistry
+     */
+    private $clientRegistry;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $em;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+
+    public function __construct(ClientRegistry $clientRegistry, EntityManagerInterface $em, RouterInterface $router)
+    {
+        $this->clientRegistry = $clientRegistry;
+        $this->em = $em;
+        $this->router = $router;
+    }
+
+    /**
+     * @param Request $request
+     * @return bool
+     */
+    public function supports(Request $request)
+    {
+        // continue ONLY if the current ROUTE matches the check ROUTE
+        return $request->attributes->get('_route') === 'connect_google_check';
+    }
+
+    /**
+     * @param Request $request
+     * @return \League\OAuth2\Client\Token\AccessToken|mixed
+     */
+    public function getCredentials(Request $request)
+    {
+        // this method is only called if supports() returns true
+
+        return $this->fetchAccessToken($this->getGoogleClient());
+    }
+
+    /**
+     * @param mixed $credentials
+     * @param UserProviderInterface $userProvider
+     * @return User|null|object|\Symfony\Component\Security\Core\User\UserInterface
+     */
+    public function getUser($credentials, UserProviderInterface $userProvider)
+    {
+        /** @var GoogleUser $googleUser */
+        $googleUser = $this->getGoogleClient()
+            ->fetchUserFromToken($credentials);
+
+
+        $consumer_email = $googleUser->getEmail();
+
+        // 1) have they logged in with Google before? Easy!
+        $existingUser = $this->em->getRepository(Consumer::class)
+            ->findOneBy(['googleId' => $googleUser->getId()]);
+
+        if (null !== $existingUser) {
+            return $existingUser;
+        }
+
+        /** @var Consumer $existingUser */
+        $existingUser = $this->em->getRepository(Consumer::class)
+            ->findOneBy(['consumer_email' => $consumer_email]);
+
+        if (null !== $existingUser) {
+            $existingUser->setConsumerEmail($googleUser->getEmail());
+            $this->em->persist($existingUser);
+            $this->em->flush();
+
+
+            return $existingUser;
+        }
+
+        // 3) Maybe you just want to "register" them by creating
+        // a User object
+        $user = new Consumer();
+        $user->setGoogleId($googleUser->getId());
+        $user->setConsumerEmail($consumer_email);
+        $user->setConsumerName($googleUser->getName());
```

Comment:
Hm..I think we could set default user role here, then we would avoid that check in entity, or is there any specific reason for current implementation?